### ISSUE TITLE
Docker registries that don't sign manifests are broken

### ISSUE THREAD

**Issue by forbushbl** [cloudfoundry/dockerapplifecycle#7]
We are updating our deployment to Diego 1.30.0 and now our Docker deployments are failing due to changes in the Docker app lifecycle. 

Here is what I see when I try to push:
```
   Creating container
   Successfully created container
   Staging...
   Staging process started ...
   Failed getting docker image by tag: missing signature key  Going to retry attempt: 1
   Failed getting docker image by tag: missing signature key  Going to retry attempt: 2
   Failed getting docker image by tag: missing signature key  Going to retry attempt: 3
   Failed getting docker image by tag: missing signature key
   Staging process failed: Exit trace for group:
   builder exited with error: failed to fetch metadata from [cf/redis] with tag [latest] and insecure registries [] due to missing signature key
   Exit status 2
   Staging Failed: STG: Exited with status 2
```
This previously worked with Diego 1.28.0

I think that this issue might be caused by the new version of libtrust. That was updated in commit https://github.com/cloudfoundry/diego-release/commit/9a1235ac444544211e275e54ff12963c12143792

When I curl my registry I can see that no signature is returned in the attribute but this seems to be supported according to the [registry schema documentation](https://docs.docker.com/registry/spec/manifest-v2-1/).

When I curl my registry at `https://my-registry-host/v2/cf/redis/manifests/latest`, I get back the following manifest. This is a registry backed by [Nexus](https://help.sonatype.com/display/NXRM3/Private+Registry+for+Docker).

```
{
  "schemaVersion": 1,
  "name": "cf/redis",
  "tag": "latest",
  "architecture": "amd64",
  "fsLayers": [
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:60872199bf5f800d707c8e237cb16ec850d911b4e408cc4e9777adb0c4daf976"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:382eae952932be17e3e19f293a03dcd46997be14270fb84563700f964dc6238a"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:a1bca8e532ecd790280384459b0f38a2afcb3e8b34ad2f44f74fc5f7fa1afcb8"
    },
    {
      "blobSum": "sha256:43be9e9f0fb9e865d48a8b5e54cd5560e70b8654b2079c296584bb831ed95270"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:f4a0d1212c38ec976de047e36c9b0f19d8427333c4db7d45d403f5839ec1c8fb"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:be9835c278527bc23a93cf0cbfae60655ad5cf9f26806935a1bd0da57d14c751"
    },
    {
      "blobSum": "sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4"
    },
    {
      "blobSum": "sha256:252a292eab15f483b2bfc2d1276bb9f358401d0ba2c23d4ce71366130da007ad"
    }
  ],
  "history": [
    {
      "v1Compatibility": "{\"created\":\"2017-10-06T23:52:36.338847492Z\",\"docker_version\":\"17.09.0-ce\",\"architecture\":\"amd64\",\"config\":{\"Hostname\":\"\",\"Domainname\":\"\",\"User\":\"\",\"AttachStdin\":false,\"AttachStdout\":false,\"AttachStderr\":false,\"ExposedPorts\":{\"6379/tcp\":{}},\"Tty\":false,\"OpenStdin\":false,\"StdinOnce\":false,\"Env\":[\"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\",\"GOSU_VERSION=1.10\",\"REDIS_VERSION=4.0.2\",\"REDIS_DOWNLOAD_URL=http://download.redis.io/releases/redis-4.0.2.tar.gz\",\"REDIS_DOWNLOAD_SHA=b1a0915dbc91b979d06df1977fe594c3fa9b189f1f3d38743a2948c9f7634813\"],\"Cmd\":[\"redis-server\",\"/usr/local/etc/redis/redis.conf\"],\"ArgsEscaped\":true,\"Image\":\"sha256:41b22f112a5fc828aaa26d2c142b02cbf71c7c99046e72f164649ac3fdc61d3b\",\"Volumes\":{\"/data\":{}},\"WorkingDir\":\"/data\",\"Entrypoint\":[\"docker-entrypoint.sh\"],\"OnBuild\":[],\"Labels\":null},\"container\":\"c5ce69e58e504b01c89aed94dd9a2865122be2aa1368a1613bb2c6b5697c28b5\",\"os\":\"linux\",\"container_config\":{\"Hostname\":\"c5ce69e58e50\",\"Domainname\":\"\",\"User\":\"\",\"AttachStdin\":false,\"AttachStdout\":false,\"AttachStderr\":false,\"ExposedPorts\":{\"6379/tcp\":{}},\"Tty\":false,\"OpenStdin\":false,\"StdinOnce\":false,\"Env\":[\"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin\",\"GOSU_VERSION=1.10\",\"REDIS_VERSION=4.0.2\",\"REDIS_DOWNLOAD_URL=http://download.redis.io/releases/redis-4.0.2.tar.gz\",\"REDIS_DOWNLOAD_SHA=b1a0915dbc91b979d06df1977fe594c3fa9b189f1f3d38743a2948c9f7634813\"],\"Cmd\":[\"/bin/sh\",\"-c\",\"#(nop) \",\"CMD [\\\"redis-server\\\" \\\"/usr/local/etc/redis/redis.conf\\\"]\"],\"ArgsEscaped\":true,\"Image\":\"sha256:41b22f112a5fc828aaa26d2c142b02cbf71c7c99046e72f164649ac3fdc61d3b\",\"Volumes\":{\"/data\":{}},\"WorkingDir\":\"/data\",\"Entrypoint\":[\"docker-entrypoint.sh\"],\"OnBuild\":[],\"Labels\":{}},\"id\":\"c3a28bd0acc7bba61dc180b3ab0e38551afaaf9d096d75904f60ff29989361bf\",\"parent\":\"779d741d2428a2fbde8b4e31e29508936c768f26fd238aef5b74aa01950fdfad\"}"
    },
    {
      "v1Compatibility": "{\"id\":\"779d741d2428a2fbde8b4e31e29508936c768f26fd238aef5b74aa01950fdfad\",\"created\":\"2017-10-06T23:52:35.703552622Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop) COPY file:2430ad853383fc9a9d4cedd163c1fec43ea775bae48e10a7b9aa64d2c1273b85 in /usr/local/etc/redis/redis.conf \"},\"parent\":\"9a29174c8e82d80e08acce0d6832c53320650eb35edceb26b42780c8fdfbff94\"}"
    },
    {
      "v1Compatibility": "{\"id\":\"9a29174c8e82d80e08acce0d6832c53320650eb35edceb26b42780c8fdfbff94\",\"created\":\"2017-09-21T20:56:10.731471356Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop)  CMD [\\\"redis-server\\\"]\"},\"parent\":\"479c2bca3cbd7538f424b3d6bbbc96bb4f5e1226bbabee7f424f410a662a9706\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"479c2bca3cbd7538f424b3d6bbbc96bb4f5e1226bbabee7f424f410a662a9706\",\"created\":\"2017-09-21T20:56:10.547593063Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop)  EXPOSE 6379/tcp\"},\"parent\":\"4cd5968c059352e00c3c9c6b0184b052da03f03d4b008c1ced6b917cafbdee14\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"4cd5968c059352e00c3c9c6b0184b052da03f03d4b008c1ced6b917cafbdee14\",\"created\":\"2017-09-21T20:56:10.363476487Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop)  ENTRYPOINT [\\\"docker-entrypoint.sh\\\"]\"},\"parent\":\"25f708a7f69ec63149d940a6f0245375230229a7fc53972715c9ca9a0679cc4f\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"25f708a7f69ec63149d940a6f0245375230229a7fc53972715c9ca9a0679cc4f\",\"created\":\"2017-09-21T20:56:10.183704175Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop) COPY file:9c29fbe8374a97f9c2d953c9c8b7224554607eeb7a610a930844f2bec678265c in /usr/local/bin/ \"},\"parent\":\"e4ef14b8627cd28c82094fda8568c4415c267486d27a06477ccde3b42d7ec5b1\"}"
    },
    {
      "v1Compatibility": "{\"id\":\"e4ef14b8627cd28c82094fda8568c4415c267486d27a06477ccde3b42d7ec5b1\",\"created\":\"2017-09-21T20:56:09.903097584Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop) WORKDIR /data\"},\"parent\":\"bb0ad9562d70386da305022c99e0158ce4bff72cc06366d8a80f7ea3fdfa0369\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"bb0ad9562d70386da305022c99e0158ce4bff72cc06366d8a80f7ea3fdfa0369\",\"created\":\"2017-09-21T20:56:09.719849147Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop)  VOLUME [/data]\"},\"parent\":\"3b579fec4c14e8988ca62cb6bbb3aa18c9dbabfccaf2332b42ab8ac8fa214c54\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"3b579fec4c14e8988ca62cb6bbb3aa18c9dbabfccaf2332b42ab8ac8fa214c54\",\"created\":\"2017-09-21T20:56:09.555145865Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c mkdir /data && chown redis:redis /data\"},\"parent\":\"ff3472e58c7c86541891d09965562365b15bc07fbe9c99a747bc7138ed33dbdd\"}"
    },
    {
      "v1Compatibility": "{\"id\":\"ff3472e58c7c86541891d09965562365b15bc07fbe9c99a747bc7138ed33dbdd\",\"created\":\"2017-09-21T20:56:08.844187657Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c set -ex; \\t\\tbuildDeps=' \\t\\twget \\t\\t\\t\\tgcc \\t\\tlibc6-dev \\t\\tmake \\t'; \\tapt-get update; \\tapt-get install -y $buildDeps --no-install-recommends; \\trm -rf /var/lib/apt/lists/*; \\t\\twget -O redis.tar.gz \\\"$REDIS_DOWNLOAD_URL\\\"; \\techo \\\"$REDIS_DOWNLOAD_SHA *redis.tar.gz\\\" | sha256sum -c -; \\tmkdir -p /usr/src/redis; \\ttar -xzf redis.tar.gz -C /usr/src/redis --strip-components=1; \\trm redis.tar.gz; \\t\\tgrep -q '^#define CONFIG_DEFAULT_PROTECTED_MODE 1$' /usr/src/redis/src/server.h; \\tsed -ri 's!^(#define CONFIG_DEFAULT_PROTECTED_MODE) 1$!\\\\1 0!' /usr/src/redis/src/server.h; \\tgrep -q '^#define CONFIG_DEFAULT_PROTECTED_MODE 0$' /usr/src/redis/src/server.h; \\t\\tmake -C /usr/src/redis -j \\\"$(nproc)\\\"; \\tmake -C /usr/src/redis install; \\t\\trm -r /usr/src/redis; \\t\\tapt-get purge -y --auto-remove $buildDeps\"},\"parent\":\"8bddd6d9f74e15f6d11b2a8d4036a6cb37f9c8ad8c0826e92ac7c42026e1c278\"}"
    },
    {
      "v1Compatibility": "{\"id\":\"8bddd6d9f74e15f6d11b2a8d4036a6cb37f9c8ad8c0826e92ac7c42026e1c278\",\"created\":\"2017-09-21T20:55:13.522915726Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop)  ENV REDIS_DOWNLOAD_SHA=b1a0915dbc91b979d06df1977fe594c3fa9b189f1f3d38743a2948c9f7634813\"},\"parent\":\"75d765a82d2aa559f9a8ecf5d43bdc72bf6711cf8095c32a43516064959e43d8\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"75d765a82d2aa559f9a8ecf5d43bdc72bf6711cf8095c32a43516064959e43d8\",\"created\":\"2017-09-21T20:55:13.339071013Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop)  ENV REDIS_DOWNLOAD_URL=http://download.redis.io/releases/redis-4.0.2.tar.gz\"},\"parent\":\"b158d249016cdd92dd93b9a3545357b9a50c975774db1154224e2546973ce7dc\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"b158d249016cdd92dd93b9a3545357b9a50c975774db1154224e2546973ce7dc\",\"created\":\"2017-09-21T20:55:13.142977463Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop)  ENV REDIS_VERSION=4.0.2\"},\"parent\":\"c14611f7247da846abd04a24c4ce1bc0e0b0931c7a5e58518a5167cec4a42b2d\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"c14611f7247da846abd04a24c4ce1bc0e0b0931c7a5e58518a5167cec4a42b2d\",\"created\":\"2017-09-19T03:32:56.931811402Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c set -ex; \\t\\tfetchDeps='ca-certificates wget'; \\tapt-get update; \\tapt-get install -y --no-install-recommends $fetchDeps; \\trm -rf /var/lib/apt/lists/*; \\t\\tdpkgArch=\\\"$(dpkg --print-architecture | awk -F- '{ print $NF }')\\\"; \\twget -O /usr/local/bin/gosu \\\"https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch\\\"; \\twget -O /usr/local/bin/gosu.asc \\\"https://github.com/tianon/gosu/releases/download/$GOSU_VERSION/gosu-$dpkgArch.asc\\\"; \\texport GNUPGHOME=\\\"$(mktemp -d)\\\"; \\tgpg --keyserver ha.pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4; \\tgpg --batch --verify /usr/local/bin/gosu.asc /usr/local/bin/gosu; \\trm -r \\\"$GNUPGHOME\\\" /usr/local/bin/gosu.asc; \\tchmod +x /usr/local/bin/gosu; \\tgosu nobody true; \\t\\tapt-get purge -y --auto-remove $fetchDeps\"},\"parent\":\"ce60d05f850f572986992f90dac46e6538955f672ab221b6ad313ee04b002a46\"}"
    },
    {
      "v1Compatibility": "{\"id\":\"ce60d05f850f572986992f90dac46e6538955f672ab221b6ad313ee04b002a46\",\"created\":\"2017-09-19T03:32:28.211706251Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop)  ENV GOSU_VERSION=1.10\"},\"parent\":\"348fef3b5a8d26c41523d050ea4b2e424eed8564616194c5f4a5a7471f650e23\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"348fef3b5a8d26c41523d050ea4b2e424eed8564616194c5f4a5a7471f650e23\",\"created\":\"2017-09-19T03:32:28.048147091Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c groupadd -r redis && useradd -r -g redis redis\"},\"parent\":\"a75874ea14dd0bd69857d039604b15ed907953b53c27b27fceb4855afb641850\"}"
    },
    {
      "v1Compatibility": "{\"id\":\"a75874ea14dd0bd69857d039604b15ed907953b53c27b27fceb4855afb641850\",\"created\":\"2017-09-13T08:40:50.323416306Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop)  CMD [\\\"bash\\\"]\"},\"parent\":\"a27cb56404e3096f806c0a68e035aed5fd479535e3bfe38750027b961e6b5557\",\"throwaway\":true}"
    },
    {
      "v1Compatibility": "{\"id\":\"a27cb56404e3096f806c0a68e035aed5fd479535e3bfe38750027b961e6b5557\",\"created\":\"2017-09-13T08:40:50.081736701Z\",\"Cmd\":{\"container_config\":\"/bin/sh -c #(nop) ADD file:bdab114a5717b42a5e02e6f602d5eeb48fc998a60d200704b4da1a7ce8552775 in / \"}}"
    }
  ]
}
```

**Comment by emalm** [comment]
Thanks for the detailed report, @forbushbl. Prioritized for the Diego team to investigate and fix.

Best,
Eric

**Comment by emalm** [comment]
Also, @forbushbl, just to confirm, are you running v3 of the Nexus repository? It appears that way from the [link you provided](https://help.sonatype.com/display/NXRM3/Private+Registry+for+Docker), but we'd like to make sure we're running the right version when we go to reproduce the issue.

Thanks again,
Eric

**Comment by forbushbl** [comment]
Thanks for giving this priority! We are running Nexus OSS 3.4.0-02

**Comment by emalm** [comment]
Thanks, @forbushbl ! That version is listed on https://hub.docker.com/r/sonatype/nexus3/tags/, so I'll suggest to the pair that picks this work to start by running that image.

**Comment by emalm** [comment]
Hi, @forbushbl,

This issue should now be fixed in the just-released [Diego v1.32.0](https://github.com/cloudfoundry/diego-release/releases/tag/v1.32.0). Please let us know if you still see this issue on that version of the release!

Best,
Eric
